src/passkey_child/passkey_child_credentials.c

```diff
@@ -90,17 +90,21 @@ prepare_credentials(struct passkey_data *data, fido_dev_t *dev,
 
     DEBUG(SSSDBG_TRACE_FUNC, "Setting user: %s.\n", data->shortname);
 
-    ret = fido_cred_set_user(cred, userid, sizeof(userid), data->shortname,
-                             NULL, NULL);
+    ret = fido_cred_set_user(cred, data->user_id, USER_ID_SIZE,
+                             data->shortname, NULL, NULL);
     if (ret != FIDO_OK) {
         DEBUG(SSSDBG_OP_FAILURE, "fido_cred_set_user failed [%d]: %s.\n",
               ret, fido_strerr(ret));
         goto done;
     }
 
+    if (data->cred_type == CRED_DISCOVERABLE) {
+        rk = FIDO_OPT_TRUE;
```

Comment:
Hi,

I would suggest to add a third `cred_type`, something like `CRED_NOT_SET` or `CRED_AUTHENTICATOR_DEFAULT`. If `--cred-type=server-side` would be given on the command line but the authenticator device is configured to use discoverable keys by default using `FIDO_OPT_OMIT` would lead to an unexpected credential type.

bye,
Sumit

Reply:
Is it possible to use discoverable keys by default? If so, how can I do it? I've been checking `ykman` command and I haven't found the way.

Reply:
Hi,

I don't know if this is possible with Yubikey devices. But I think we should make sure server-side credentials are generate if requested and I think the only way to make sure is to use `FIDO_OPT_FALSE`.

bye,
Sumit

Reply:
We can't do that. For compatibility reasons with keys that don't support discoverable credentials we need to set it to `FIDO_OPT_OMIT`. [Link to explanation](https://github.com/SSSD/sssd/pull/6327#discussion_r961424250).